components/sensor/ultrasonic/ultrasonic.go

```diff
@@ -130,16 +126,28 @@ func (s *Sensor) namedError(err error) error {
 func (s *Sensor) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.echoInterrupt.AddCallback(s.ticksChan)
-	defer s.echoInterrupt.RemoveCallback(s.ticksChan)
+
+	// Grab the 2 pins from the board. We don't just get these once during setup, in case the board
+	// reconfigures itself because someone decided to rewire things.
+	echoInterrupt, ok := s.board.DigitalInterruptByName(s.config.EchoInterrupt)
+	if !ok {
+		return nil, errors.Errorf("ultrasonic: cannot grab digital interrupt %q", s.config.EchoInterrupt)
+	}
+	triggerPin, err := s.board.GPIOPinByName(s.config.TriggerPin)
```

Comment:
I'd rather we move away from storing the entire Config in the component struct, makes reconfiguration logic when we implement it a little tricky. This component has an always rebuild flow so okay, but that is my preference overall.

Reply:
Oh! Anything that stores its config cannot be `resource.TriviallyReconfigurable`. If storing the config is the only reason it's not trivial, then I agree that we should consider ways to stop storing the config.